src/instruments.rs

```diff
@@ -1,133 +1,331 @@
 //! interfacing with the `instruments` command line tool
 
 use std::fs;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 use std::process::{Command, Output};
 
 use anyhow::{anyhow, Result};
+use cargo::core::Workspace;
+use semver::Version;
 
-use crate::opt::Opts;
+use crate::opt::AppConfig;
 
-/// Check that `instruments` is in $PATH.
-pub(crate) fn check_existence() -> Result<()> {
-    let path = ["/", "usr", "bin", "instruments"].iter().collect::<PathBuf>();
-    if path.exists() {
-        Ok(())
-    } else {
+/// Holds available templates.
+///
+pub struct TemplateCatalog {
+    standard_templates: Vec<String>,
+    custom_templates: Vec<String>,
+}
+
+impl TemplateCatalog {
+    /// Return a vector of all templates.
+    fn all_templates(&self) -> Vec<&str> {
+        self.standard_templates
+            .iter()
+            .chain(self.custom_templates.iter())
+            .map(|value| value.as_str())
+            .collect::<Vec<_>>()
+    }
+}
+
+/// Represents the Xcode Instrument version detected.
+pub enum XcodeInstruments {
```

Comment:
Is this `pub` in a meaningful way? Since this is a `bin` crate I suppose it doesn't really matter? Otherwise I would suggest `pub(crate)`.

Reply:
No, no there was no meaningful reason. Thanks

Reply:
Eh, I replied too quickly, `XcodeInstruments` is used from `app.rs`.